source/components/JsxParser.tsx

```diff
@@ -103,25 +100,27 @@ export default class JsxParser extends React.Component<TProps> {
 			case 'BinaryExpression':
 				const binaryLeft = this.#parseExpression(expression.left, scope)
 				const binaryRight = this.#parseExpression(expression.right, scope)
-				/* eslint-disable eqeqeq,max-len */
+				let binaryResult
 				switch (expression.operator) {
-					case '-': return binaryLeft - binaryRight
-					case '!=': return binaryLeft != binaryRight
-					case '!==': return binaryLeft !== binaryRight
-					case '*': return binaryLeft * binaryRight
-					case '**': return binaryLeft ** binaryRight
-					case '/': return binaryLeft / binaryRight
-					case '%': return binaryLeft % binaryRight
-					case '+': return binaryLeft + binaryRight
-					case '<': return binaryLeft < binaryRight
-					case '<=': return binaryLeft <= binaryRight
-					case '==': return binaryLeft == binaryRight
-					case '===': return binaryLeft === binaryRight
-					case '>': return binaryLeft > binaryRight
-					case '>=': return binaryLeft >= binaryRight
-					/* eslint-enable eqeqeq,max-len */
+					case '-': binaryResult = binaryLeft - binaryRight; break
+					case '!=': binaryResult = binaryLeft != binaryRight; break // eslint-disable-line eqeqeq
+					case '!==': binaryResult = binaryLeft !== binaryRight; break
+					case '*': binaryResult = binaryLeft * binaryRight; break
+					case '**': binaryResult = binaryLeft ** binaryRight; break
+					case '/': binaryResult = binaryLeft / binaryRight; break
+					case '%': binaryResult = binaryLeft % binaryRight; break
+					case '+': binaryResult = binaryLeft + binaryRight; break
+					case '<': binaryResult = binaryLeft < binaryRight; break
+					case '<=': binaryResult = binaryLeft <= binaryRight; break
+					case '==': binaryResult = binaryLeft == binaryRight; break // eslint-disable-line eqeqeq
+					case '===': binaryResult = binaryLeft === binaryRight; break
+					case '>': binaryResult = binaryLeft > binaryRight; break
+					case '>=': binaryResult = binaryLeft >= binaryRight; break
+					default:
+						this.props.onError!(new Error(`Unsupported binary operator: ${expression.operator}`))
+						return undefined
 				}
-				return undefined
+				return handleNaN(binaryResult)
```

Comment:
this helper ensures that we actually treat `NaN` as a real value and output its string version when appropriate.